docs/plugins.py

```diff
@@ -0,0 +1,110 @@
+import asyncio
+
+import aiohttp
+import pandas as pd
+import yaml
+
+
+def format_package_links(package_name, repo_link):
+    return f'<a href="{repo_link}">{package_name}</a>'
+
+
+def format_repo_link(repo_link):
+    if "http" not in repo_link:
+        return f"https://github.com/{repo_link}/"
+    return repo_link
+
+
+def format_badge_html(badge, link):
+    return f'<a href="{link}"><img src="{badge}"/></a>'
+
+
+async def check_ok(client, url):
+    async with client.get(url) as r:
+        if "anaconda.org" in url:
+            body = await r.text()
+            if "requires authentication" in body:
+                return False
+        return r.ok
+
+
+async def check_all_ok(urls):
+    async with aiohttp.client.ClientSession() as c:
+        coroutines = [check_ok(c, url) for url in urls]
+        return await asyncio.gather(*coroutines)
+
+
+def generate_plugin_table():
+    plugins = yaml.safe_load(open("plugins.yaml", "rb"))
+    plugin_df = pd.DataFrame(plugins)
+    plugin_df = plugin_df.rename(columns={"description": "Description", "drivers": "Drivers"})
+
+    plugin_df["short_name"] = plugin_df["name"].apply(lambda x: x.split("/")[-1])
+    plugin_df["repo_links"] = plugin_df["repo"].apply(format_repo_link)
+    plugin_df["conda_package"] = plugin_df["conda_package"].fillna(plugin_df["short_name"])
+
+    # CI badges
+    plugin_df["ci_badges"] = plugin_df["repo_links"].apply(lambda x: f"{x}/workflows/CI/badge.svg")
+    plugin_df["ci_links"] = plugin_df["repo_links"].apply(lambda x: f"{x}/actions")
+    ci_badges_ok = asyncio.run(check_all_ok(plugin_df["ci_badges"]))
+
+    # Docs badges
+    plugin_df["docs_badges"] = plugin_df["short_name"].apply(lambda x: f"https://readthedocs.org/projects/{x}/badge/?version=latest")
+    plugin_df["docs_links"] = plugin_df["short_name"].apply(lambda x: f"https://{x}.readthedocs.io/en/latest/?badge=latest")
+    docs_badges_ok = asyncio.run(check_all_ok(plugin_df["docs_links"]))
+
+    # PyPi badges
+    plugin_df["pypi_badges"] = plugin_df["short_name"].apply(lambda x: f"https://img.shields.io/pypi/v/{x}.svg?maxAge=3600")
+    plugin_df["pypi_links"] = plugin_df["short_name"].apply(lambda x: f"https://pypi.org/project/{x}")
+    pypi_badges_ok = asyncio.run(check_all_ok(plugin_df["pypi_links"]))
+
+    # Conda badges
```

Comment:
We probably only need conda defaults and conda-forge; any that aren't on one of those should be migrated.